src/torchcodec/_core/CudaDeviceInterface.cpp

```diff
@@ -224,41 +224,66 @@ void CudaDeviceInterface::convertAVFrameToFrameOutput(
   // Use the user-requested GPU for running the NPP kernel.
   c10::cuda::CUDAGuard deviceGuard(device_);
 
+  cudaStream_t rawStream = at::cuda::getCurrentCUDAStream().stream();
+
+  // Build an NppStreamContext, either via the old helper or by hand on
+  // CUDA 12.9+
+  NppStreamContext nppCtx{};
+#if CUDA_VERSION < 12090
+  NppStatus ctxStat = nppGetStreamContext(&nppCtx);
+  TORCH_CHECK(ctxStat == NPP_SUCCESS, "nppGetStreamContext failed");
+  // override if you want to force a particular stream
+  nppCtx.hStream = rawStream;
+#else
+  // CUDA 12.9+: helper was removed, we need to build it manually
+  int dev = 0;
+  cudaError_t err = cudaGetDevice(&dev);
```

Comment:
Hi @Kh4L , I actually have some questions before moving forward:

1. Should we just rely on the existing `device_`  attribute instead of calling `cudaGetDevice(&dev)`, or are they actually equivalent?
2. Would it make sense to cache the `nppCtx` across calls? In this PR it looks like we're creating the context over and over for every single frame that needs to be decoded. I wonder if it might be beneficial to cache it in the class and re-use it?

Thanks for your help so far, I'm still trying to build familiarity with that part of the code base.